kb/prometheus-grafana-setup-cloud.md

```diff
@@ -0,0 +1,252 @@
+
+For this example, you can run Prometheus locally, or as a docker container. In either case, ensure that you set the listen targets to the ports where you have exposed your scrape endpoints.
+When running Prometheus locally, set your target address to port 8077 that we set in the SDK metrics setup example, in your Prometheus configuration YAML file.
+
+Example:
+
+```
+global:
+ scrape_interval: 10s # Set the scrape interval to every 10 seconds. Default is every 1 minute.
+#...
+
+# set your scrape configuration targets to the ports exposed on your endpoints in the SDK
+scrape_configs:
+ - job_name: 'temporalsdkmetrics'
+   metrics_path: /metrics
+   scheme: http
+   static_configs:
+     - targets:
+       # This is the scrape endpoint where Prometheus listens for SDK metrics.
+       - localhost:8077
+       # You can have multiple targets here, provided they are set up in your application code.
+
+```
+
+See the Prometheus documentation for more details on how you can run Prometheus locally or using docker.
+
+Note that Temporal Cloud exposes metrics through a Prometheus HTTP API endpoint (not a scrape endpoint) that can be configured as a datasource in Grafana. The Prometheus configuration described here is for scraping metrics data on endpoints for SDK metrics only.
+
+To check whether Prometheus is receiving metrics from your SDK target, go to http://localhost:9090 and navigate to Status > Targets. The status of your target endpoint defined in your configuration will show here.
+
+## Data sources configuration for Temporal Cloud and SDK metrics in Grafana
+
+Depending on how you use Grafana, you can either install and run it locally, run it as a docker container, or log in to Grafana Cloud to set up your data sources.
+
+If you have installed and are running Grafana locally, go to [http://localhost:3000](http://localhost:3000) and sign in.
+
+You must configure your Temporal Cloud and the SDK Metrics data sources separately in Grafana.
+
+To add the Temporal Cloud Prometheus HTTP API endpoint that we generated in the previous section, do the following:
+
+1. Go to **Configuration** > **Data sources**.
+2. Select **Add data source** > **Prometheus**.
+3. Enter a name for your Temporal Cloud metrics data source, for example _Temporal Cloud metrics_.
+4. In the **HTTP section**, paste the URL that was generated in the Observability section on the Temporal Cloud UI.
+5. In the **Auth section**, enable **TLS Client Auth**.
+6. In the **TLS/SSL Auth Details** section, paste the root CA certificate and key that you used to [set up Observability](#temporal-cloud-metrics-setup) in your Temporal Cloud UI.
+7. Select **Save and test** to ensure the data source is working properly.
+
+If you see issues in setting this data source, verify that you are using the correct certificates (root CA) on both your Temporal Cloud Observability setup and in the TLS authentication in Grafana.
```

Comment:
> If you see issues in setting this data source, verify that you are using the correct certificates (root CA) on both your Temporal Cloud Observability setup and in the TLS authentication in Grafana.

I don't think you need the CA to setup Grafana. 🤔

Reply:
Dang you're right; I used end-entity certs not root in my Grafana. Will update this.

Reply:
yeah! thanks for pointing that out. I used client certs in Grafana but wrote this incorrectly here. Will update it.